File: src/main/scala/com/dyonovan/neotech/common/container/machines/ContainerElectricCrusher.scala
package com.dyonovan.neotech.common.container.machines

import com.dyonovan.neotech.common.tiles.machines.TileElectricCrusher
import com.teambr.bookshelf.common.container.BaseContainer
import net.minecraft.entity.player.InventoryPlayer
import net.minecraft.inventory.SlotFurnaceOutput

/**
 * This file was created for NeoTech
 *
 * NeoTech is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Dyonovan
 * @since August 13, 2015
 */
class ContainerElectricCrusher(playerInventory: InventoryPlayer, tile: TileElectricCrusher) extends
        BaseContainer(playerInventory, tile) {

    addSlotToContainer(new RestrictedSlot(tile, 0, 41, 35))
    addSlotToContainer(new SlotFurnaceOutput(playerInventory.player, tile, 1, 101, 35))
    addSlotToContainer(new SlotFurnaceOutput(playerInventory.player, tile, 2, 131, 35))
    addPlayerInventorySlots(8, 84)
}

File: src/main/java/com/teambrmodding/neotech/common/container/machines/processors/ContainerElectricCrusher.java
package com.teambrmodding.neotech.common.container.machines.processors;

import com.teambrmodding.neotech.common.container.machines.ContainerAbstractMachine;
import com.teambrmodding.neotech.common.tiles.AbstractMachine;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.item.ItemStack;
import net.minecraftforge.items.SlotItemHandler;

/**
 * This file was created for NeoTech
 * <p>
 * NeoTech is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis - pauljoda
 * @since 2/15/2017
 */
public class ContainerElectricCrusher extends ContainerAbstractMachine {

    /**
     * Creates the object
     *
     * @param playerInventory The players inventory
     * @param inventory       The tile/object inventory
     */
    public ContainerElectricCrusher(InventoryPlayer playerInventory, AbstractMachine inventory) {
        super(playerInventory, inventory);

        addSlotToContainer(new SlotItemHandler(inventory, 0, 56, 35));
        addSlotToContainer(new SlotItemHandler(inventory, 1, 117, 35) {
            /**
             * Check if the stack is a valid item for this slot. Always true beside for the armor slots.
             */
            @Override
            public boolean isItemValid(ItemStack stack) {
                return false;
            }
        });
        addSlotToContainer(new SlotItemHandler(inventory, 2, 139, 35) {
            /**
             * Check if the stack is a valid item for this slot. Always true beside for the armor slots.
             */
            @Override
            public boolean isItemValid(ItemStack stack) {
                return false;
            }
        });

        addPlayerInventorySlots(84);
    }
}
